### DISCUSSION TITLE
executeQuery() not working.

### DISCUSSION THREAD

**Discussion by Manubi** [urql-graphql/urql#1836] [Q&A] [answered]
Hey,
Maybe someone has an idea why `executeQuery()` is not working?
<img width="892" alt="CleanShot 2021-08-02 at 12 47 52@2x" src="https://user-images.githubusercontent.com/20935138/127850305-f1b0cc49-703b-49b7-8da0-c727f62fafab.png">
![CleanShot 2021-08-02 at 12 51 00](https://user-images.githubusercontent.com/20935138/127850523-64370dfd-51fb-420a-b51f-9174c24f4db6.png)

<img width="1057" alt="CleanShot 2021-08-02 at 12 49 41@2x" src="https://user-images.githubusercontent.com/20935138/127850336-8b454994-1304-44c4-945c-3b1c0aa5a91d.png">

Thanks in advance!

**Comment by JoviDeCroock** [comment, marked as answer]
It is working but you've dispatched it with the `requestPolicy: 'cache-first'`, try doing this `executeQuery({ requestPolicy: 'network-only' })`. Like [this](https://formidable.com/open-source/urql/docs/basics/react-preact/#reexecuting-queries)

**Reply by Manubi** [reply]
Really love your fast support! 

Just to be sure. Is this behaviour wanted? I mean don't get me wrong, but if I want to re-execute the query I normally want to get the latest state from the server. And I was under the impression, that cache-first just means, look into cache and revalidate in the background?

Thanks!

**Reply by JoviDeCroock** [reply]
I guess there is something to be said about the unexpected nature of this, it follows the default `requestPolicy` set on your Client. If the user opts for executing with `cache-and-network` this is fully possible. I'm going to throw this into our chat-group as well as I see how this could be confusing

> that cache-first just means, look into cache and revalidate in the background?

No this is cache-and-network, cache-first will not reach over the network if there's a _full_ cache-result

**Reply by Manubi** [reply]
Somehow an info that it "just" fetched the cache would help a lot. Because right now with this, I had the impression that the code is not working and nothing happens.

**Reply by JoviDeCroock** [reply]
All right, yes, I'll see about adding a note about this using the `client.requestPolicy` by default. Thank you for the feedback! Much appreciated
